### PR TITLE
[FIX] l10n_cl: unable to issue credit note

### PR DESCRIPTION
**Pull request by agr-odoo**
Have a CL company setup
Issue an invoice
Then generate the credit note via 'Add Credit Note' wizard
Add a reason (credit method is locked to 'partial refund') and confirm
Error will raise “You can not use a invoice document type with a refund
invoice”

This occur after
https://github.com/odoo/enterprise/commit/9e049b833b23809e4287ef61ddf7482413debefd

We now fall back to the logic of l10n_latam_invoice_document
which, in turns, fall back to the l10n_cl, ignoring the 'credit_note'
case.

opw-2794515

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
@jco-odoo 
Not sure about this

After the mentioned fix https://github.com/odoo/enterprise/commit/9e049b833b23809e4287ef61ddf7482413debefd
We compute the document type here https://github.com/odoo/odoo/blob/e5a76880320ce0606e6a0fc2a6ba776f79ce126e/addons/l10n_latam_invoice_document/models/account_move.py#L209
But instead of using the `_get_l10n_latam_documents_domain` just above, it is overridden by `l10n_cl`
https://github.com/odoo/odoo/blob/e5a76880320ce0606e6a0fc2a6ba776f79ce126e/addons/l10n_cl/models/account_move.py#L18

Like in MX there is a lot of domain specifc knowledge involved with these modules, so I would like to know if this change is feasible/safe

**Comment by jco-odoo** [comment]
@ren-odoo What happens if you create a credit note from scratch in v14?  Does it find a document type by default or not?  This error has more to do with the domain itself than with a fix for the change in the wizard.

**Comment by agr-odoo** [comment]
@jco-odoo 
The domain is the same both for credit note created from scratch or created from the reverse move wizard
`[('country_id.code', '=', 'CL'), ('internal_type', '!=', 'invoice_in'), ('code', '!=', '71')]`
This will return several records in `l10n_latam_available_document_type_ids`
So users will be able to choose the document type when creating the credit note from scratch
The problem is in the reverse move wizard, where the choice is hidden/fixed, so it will stick to the first one retrieved, which is the invoice type

**Comment by ren-odoo** [comment]
Hi @jco-odoo 

Without this fix in V14 when you create a credit note from scratch the document type assign by defult is Electronic Customer invoice. 

I tested this PR and the credit note creation using the wizard is working correctly for three options and the document type 61- Electronic Credit note is correctly assigned, I also tested Credit Notes created from scratch and the deafult document type now is 61 - Electronic Credit Note. 

@Danisan can you take a look at this PR as well, please? 

cc. @vbe-odoo @fvz-odoo

File: addons/l10n_cl/models/account_move.py [hunk 1]
```diff
@@ -21,7 +21,11 @@ def _get_l10n_latam_documents_domain(self):
                 self.journal_id.l10n_latam_use_documents:
             return super()._get_l10n_latam_documents_domain()
         if self.journal_id.type == 'sale':
-            domain = [('country_id.code', '=', "CL"), ('internal_type', '!=', 'invoice_in')]
+            if self.move_type in ['out_refund', 'in_refund']:
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py:24]: There should not be any in_refund here normally.

File: addons/l10n_cl/models/account_move.py [hunk 2]
```diff
@@ -21,7 +21,11 @@ def _get_l10n_latam_documents_domain(self):
...
+                internal_types_domain = ('internal_type', '=', 'credit_note')
+            else:
+                internal_types_domain = ('internal_type', '!=', 'invoice_in')
+            domain = [('country_id.code', '=', "CL"), internal_types_domain]
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py:28]: maybe use '' everywhere then

**Comment by jco-odoo** [comment]
@agr-odoo Would be nice if you could do those last changes.  This seems to fix it!

**Comment by agr-odoo** [comment]
Thank you all for the review and the suggestions, I've updated the patch

File: addons/l10n_cl/models/account_move.py [hunk 3]
```diff
@@ -21,7 +21,11 @@ def _get_l10n_latam_documents_domain(self):
                 self.journal_id.l10n_latam_use_documents:
             return super()._get_l10n_latam_documents_domain()
         if self.journal_id.type == 'sale':
-            domain = [('country_id.code', '=', "CL"), ('internal_type', '!=', 'invoice_in')]
+            if self.move_type == 'out_refund':
+                internal_types_domain = ('internal_type', '=', 'credit_note')
+            else:
+                internal_types_domain = ('internal_type', '!=', 'invoice_in')
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py:27]: @Danisan should not the credit note not be visible in the regular invoice?  
```suggestion
                internal_types_domain = ('internal_type', 'not in', ['invoice_in', 'credit_note'])
```

**Comment by ren-odoo** [comment]
Hi @jco-odoo 

I checked with @Danisan and he agrees on hiding the credit note document type on regular customer invoices (out_invoices) and only leave it visible on Credit notes (out_refund). 

cc. @vbe-odoo @fvz-odoo

**Comment by jco-odoo** [comment]
@agr-odoo Could you apply my suggestion then?  (and check if it is correct)

**Comment by agr-odoo** [comment]
@jco-odoo 
Done, it looks ok to me, at least from what I can observe:
|move type|available document type|
|--------------|--------------------------------|
|Invoice (out_invoice) | '(33) Factura Electrónica', '(34) Factura no Afecta o Exenta Electrónica', '(56) Nota de Débito Electrónica', '(39) Boleta Electrónica' |
|Credit note (out_refund) | '(61) Nota de Crédito Electrónica' |
|Bill (in_invoice) | '(33) Factura Electrónica', '(34) Factura no Afecta o Exenta Electrónica', '(61) Nota de Crédito Electrónica', '(56) Nota de Débito Electrónica' |
|Refund (in_refund) | '(33) Factura Electrónica', '(34) Factura no Afecta o Exenta Electrónica', '(61) Nota de Crédito Electrónica', '(56) Nota de Débito Electrónica' |

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**